File: mpp.model/src/main/java/model/ModelUser.java
package model;

import configuration.ConfigurationFactory;
import configuration.ConfigurationProtocol;
import configuration.ConfigurationType;
import domain.User;
import repository.RepositoryFactory;
import repository.RepositoryType;

/**
 * Name:        {ClassName}
 * Effect:      {ClassEffect}
 * Date:        01/04/2017
 * Tested:      False
 *
 * @author Alexandru Stoica
 * @version 1.0
 */

public class ModelUser extends Model<User, Integer> {

    public ModelUser(String database) {
        this.preferredDatabaseSource = database;
        this.checkDatabase();
        ConfigurationProtocol configuration =
                ConfigurationFactory.build(ConfigurationType.USER);
        this.repository = RepositoryFactory.getRepository(
                RepositoryType.USER,
                preferredDatabaseSource,
                configuration
        );
    }

    public ModelUser() {
        this(database);
    }
}

File: Database System Management/ModelComponent/Model/ModelUser.cs
using ModelComponent.Domain;
using ModelComponent.Entity;
using ModelComponent.Repository;

namespace ModelComponent.Model
{
    public class ModelUser : Model<User, UserEntity>
    {
        public ModelUser(string database) : base(database)
        {
            Repository = RepositoryFactory
                .GetInstance<User, UserEntity>(PreferredDatabaseSource, RepositoryType.User);
        }

        public ModelUser()
        {
            Repository = RepositoryFactory
                .GetInstance<User, UserEntity>(PreferredDatabaseSource, RepositoryType.User);
        }
    }
}